src/services/search.ts

```diff
@@ -0,0 +1,29 @@
+// parent search. holds logic for splitting between different services
+
+import * as play from 'play-dl';
+import { PlayValidation } from '../types/definitions';
+
+// TODO: handle search for other sources. for now, default to yt_videos only
+export const search = async (
+    query: string,
+): Promise<play.YouTubeVideo | null> => {
+    const validationResult: PlayValidation = await play.validate(query);
+
+    if (validationResult === 'search') {
+        const searchResult = await play.search(query, {
```

Comment:
`const [searchResult] = await play.search(...)`
if you get empty array, searchResult will be `undefined`, which you can return directly if you take my use-undefined suggestion. Otherwise, you would be able to then return `searchResult ?? null` to convert to null.

I suggest this so you can get rid of the if statement below on line 18. If statements bad!

Reply:
Cool! Didn't know you could do this!